modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -762,57 +837,93 @@ sub apply_tweaks {
                         $need_write = 1;
                     }
                 } else {
-                    print "Tweak.Error   \tAnalysis does not support '$attrib_name' attribute\n";
+                    if($operator eq '?') {
+                        $tweakStructure->{Action} = "SHOW";
+                    } elsif($operator eq '#') {
+                          $tweakStructure->{Action} = "DELETE";
+                    } else {
+                        $tweakStructure->{Action} = "SET";
+                    }
+                    $tweakStructure->{Error} = 'Field not recognized';
+                    push @response, "Tweak.Error  \tAnalysis does not support '$attrib_name' attribute\n";
                 }
+
+                push @{$responseStructure->{Tweaks}}, $tweakStructure;
             }
 
         } elsif($tweak=~/^resource_class\[([^\]]+)\]\.(\w+)(\?|=(.+))$/) {
             my ($rc_pattern, $meadow_type, $operator, $new_value_str) = ($1, $2, $3, $4);
 
             my $resource_classes = $self->collection_of( 'ResourceClass' )->find_all_by_pattern( $rc_pattern );
-            print "Tweak.Found   \t".scalar(@$resource_classes)." resource_classes matching the pattern '$rc_pattern'\n";
+            push @response, "Tweak.Found  \t".scalar(@$resource_classes)." resource_classes matching the pattern '$rc_pattern'\n";
 
             if($operator eq '?') {
                 foreach my $rc (@$resource_classes) {
+                    my $tweakStructure;
+                    $tweakStructure->{Object}->{Type} = "Resource class";
                     my $rc_name = $rc->name;
+                    $tweakStructure->{Object}->{Id} = $rc->dbID;
+                    $tweakStructure->{Object}->{Name} = $rc_name;
+                    $tweakStructure->{Action} = "SHOW";
 
                     if(my $rd = $self->collection_of( 'ResourceDescription' )->find_one_by('resource_class', $rc, 'meadow_type', $meadow_type)) {
                         my ($submission_cmd_args, $worker_cmd_args) = ($rd->submission_cmd_args, $rd->worker_cmd_args);
-                        print "Tweak.Show    \tresource_class[$rc_name].$meadow_type ::\t".stringify([$submission_cmd_args, $worker_cmd_args])."\n";
+                        push @response, "Tweak.Show  \tresource_class[$rc_name].$meadow_type ::\t".stringify([$submission_cmd_args, $worker_cmd_args])."\n";
+                        $tweakStructure->{Return}->{OldValue} = stringify([$submission_cmd_args, $worker_cmd_args]);
                     } else {
-                        print "Tweak.Show    \tresource_class[$rc_name].$meadow_type ::\t(missing values)\n";
+                        push @response, "Tweak.Show  \tresource_class[$rc_name].$meadow_type ::\t(missing values)\n";
+                        $tweakStructure->{Return}->{OldValue} = undef;
                     }
+                    $tweakStructure->{Return}->{Field} = "resource_class[$rc_name].$meadow_type";
```

Comment:
In this case, _Type_ is "Resource class" and _Name is `$rc_name`. I think it's redundant to add them in _Field_ and would set the latter to just `$meadow_type`. This would be consistent with the way analyses are modelled, e.g. `resource_class[urgent].LSF` vs `analysis[1].is_excluded`